ethereum/blockwatch/client.go

```diff
@@ -17,30 +16,34 @@ import (
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
 )
 
-var (
+const (
 	// We give up on ETH RPC requests sent for the purpose of block watching after 10 seconds
-	requestTimeout = 10 * time.Second
+	requestTimeout           = 10 * time.Second
+	bigIntParsingErrorString = "Failed to parse big.Int value from hex-encoded %s returned from %s"
 )
 
 // Client defines the methods needed to satisfy the client expected when
 // instantiating a Watcher instance.
 type Client interface {
-	HeaderByNumber(number *big.Int) (*types.MiniHeader, error)
-	HeaderByHash(hash common.Hash) (*types.MiniHeader, error)
-	FilterLogs(q ethereum.FilterQuery) ([]ethtypes.Log, error)
+	HeaderByNumber(ctx context.Context, number *big.Int) (*types.MiniHeader, error)
```

Comment:
The only reason to pass in a separate context for each of these methods would be if we needed more fine-grained control over it. For example, if we wanted to cancel the context _independent_ of the main context that is created in the `core` package (or in __cmd/mesh__). I don't see any reason why we would want to do that.

Instead, can we add a `ctx` field to `RpcClient` and use that for all requests? That would significantly reduce the overhead and still behave the same way.

Reply:
The documentation for context recommends not to do this: "Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx..." https://golang.org/pkg/context/

Reply:
Adding context in a separate PR

Reply:
@z2trillion this is one case where I do not agree with the documentation/recommendation.